common/mam/v1/merkle.h

```diff
@@ -0,0 +1,24 @@
+#ifndef __MAM_MERKLE_H__
+#define __MAM_MERKLE_H__
+
+#include "common/curl-p/trit.h"
+
+#define NULL_SIBLING 1
+#define NULL_TREE 2
+#define LEAF_INDEX_OUT_OF_BOUNDS 3
+#define DEPTH_OUT_OF_BOUNDS 4
+
+size_t merkle_size(size_t leaf_count);
+size_t merkle_depth(size_t node_count);
+size_t merkle_node_index(size_t depth, size_t width, size_t tree_depth);
+size_t merkle_leaf_index(size_t leaf_index, size_t leaf_count);
+
+int merkle_create(trit_t *const tree, trit_t *const seed, int64_t offset,
```

Comment:
Could you add documentation here? Just steal from https://github.com/iotaledger/MAM/blob/master/mam/src/mam.rs